Tennis-scheduler/src/main/java/com/tennis/tennisscheduler/controllers/TimeslotController.java

```diff
@@ -84,9 +121,29 @@ public ResponseEntity<TimeslotResponseDto> update(@PathVariable long id, @Reques
     public ResponseEntity<HttpStatus> deleteById(@PathVariable long id){
         Timeslot timeslotExisting = timeslotService.getById(id);
         if (timeslotExisting == null) {
-            return new ResponseEntity<>( HttpStatus.BAD_REQUEST);
+            return new ResponseEntity<>( HttpStatus.NOT_FOUND);
         }
         timeslotService.deleteById(id);
         return new ResponseEntity<>(HttpStatus.OK);
     }
+    @PreAuthorize("hasAnyRole('ADMIN','TENNIS_PLAYER')")
+    @PutMapping(value = "/cancel/{id}")
+    public ResponseEntity<HttpStatus> cancelTimeslot(@PathVariable long id){
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        Person user = (Person)authentication.getPrincipal();
+
+        Timeslot timeslotExisting = timeslotService.getById(id);
+        if (timeslotExisting == null)
+            return new ResponseEntity<>( HttpStatus.NOT_FOUND);
+
+        if (user.getRole().getRoleName().equals(UserType.ROLE_TENNIS_PLAYER) && timeslotExisting.getPerson().getId() != user.getId())
+            return new ResponseEntity<>( HttpStatus.UNAUTHORIZED);
+
+        if (user.getRole().getRoleName().equals(UserType.ROLE_TENNIS_PLAYER))
```

Comment:
Why do you need this if?

Reply:
And you are already checking if logged in user is different from the user in timeslot. Which means that if it is different it will fail and return 401 before it comes to this part of code.